packages/datagateway-dataview/src/views/datafilePreview/datafilePreviewer.component.tsx

```diff
@@ -0,0 +1,280 @@
+ * and also the file extension of the datafile, so that if they want to request
+ * support for a certain datafile type, they can just pass along the extension.
+ * Otherwise, it delegates the parsing and rendering of the preview to the appropriate component.
+ *
+ * The delegation is defined by a lookup table in `./previewComponents/previewComponents`.
+ * To add support for a new file extension, simply add an entry to the table,
+ * mapping the new file extension to the corresponding React component that
+ * can render a preview of the datafile.
+ * To see an example of a preview component, check out {@link TxtPreview}
+ * in `./previewComponents/txtPreview.component.tsx`.
+ *
+ * Preview components will receive the current {@link Datafile} and the content of it
+ * stored as a {@link Blob}. Since the previewer supports zoom control,
+ * they should respond to the current zoom level and adjust the size of the preview
+ * content accordingly. For example, the .txt preview component changes the font size
+ * according to the current zoom level.
+ *
+ * Currently, the previewer is only enabled for ISIS, as they don't have a 2-level system
+ * (no tape storage), and so datafiles are available immediately.
+ *
+ * @param datafileId The ID of the datafile to be previewed.
+ */
+function DatafilePreviewer({
+  datafileId,
+}: DatafilePreviewerProps): JSX.Element {
+  const [status, setStatus] = React.useState<PreviewerStatus>({
+    // the previewer will load the metadata of the datafile first
+    code: 'LOADING_METADATA',
+  });
+
+  const {
+    data: datafile,
+    isLoading: isLoadingMetadata,
+    error: loadDatafileMetaError,
+  } = useDatafileDetails(datafileId, undefined, {
+    enabled: !Number.isNaN(datafileId),
+  });
+
+  const datafileExtension = datafile && extensionOf(datafile);
+  const supportsExtension =
+    datafileExtension && isExtensionSupported(datafileExtension);
+
+  const { data: datafileContent, error: loadDatafileContentError } =
+    useDatafileContent({
+      datafileId: datafile?.id ?? -1,
+      enabled:
+        // only fetch datafile if datafile meta is available & if the extension is supported
+        Boolean(datafile) &&
+        Boolean(datafileExtension) &&
+        Boolean(supportsExtension),
+      onDownloadProgress: (event) => {
+        setStatus({
+          code: 'LOADING_CONTENT',
+          progress: (event.loaded / event.total) * 100,
+        });
+      },
+    });
+
+  const [t] = useTranslation();
+  const theme = useTheme();
+  const isDetailsPaneShown = useSelector<StateType, boolean>(
+    (state) => state.dgdataview.isisDatafilePreviewer.isDetailsPaneShown
+  );
+  const [isDetailsPaneGridVisible, setIsDetailsPaneGridVisible] =
+    React.useState(isDetailsPaneShown);
+  const [isDetailsPaneIn, setIsDetailsPaneIn] =
+    React.useState(isDetailsPaneShown);
+
+  React.useEffect(() => {
+    // This effect controls the appearance of details panel
```

Comment:
Is there not an out of the box way to do this via MUI like using `Drawer`? I have examples in OperationsGateway in how to make it not fill the whole screen/fit the container it's in. However, if this works it works, just is a little confusing having 3 variables for the details pane visibility 😅

Reply:
`Drawer` isn't quite what I'm looking for - it is full height, and it sits on top of the content, not side by side.

Reply:
It can be used like this though - see:

![OG-plot-config](https://user-images.githubusercontent.com/22525471/186659758-190315b9-2e52-40d1-b85c-092ac5a1b964.gif)

This is on OG develop. However, the current way works as well - it was just a thought

Reply:
Interesting. This may sound nitpicky, but when you expand/close the side panel, the content on the right hand side "jumps" instead of expand/shrink itself smoothly, which is why I needed to code in special logic to handle it.

Reply:
Hmm true, maybe we should steal your code in OG ;)

Reply:
Feel free to ~~steal~~ adapt it as you wish!